src/org/opensolaris/opengrok/configuration/Group.java

```diff
@@ -37,6 +37,7 @@
 
     private String name;
     private String pattern = "";
+    private Pattern compiledPattern = Pattern.compile("(" + this.pattern + ")");
```

Comment:
The empty pattern is the original design - no project matches this group (it still can be used as a superior group for several other subgroups). With this in mind there must be something like "empty" pattern - could be achieved with
1. several ifs in the code for the `pattern.isEmpty()` or `compiledPattern == null` or
2. setting an empty pattern avoiding the ifs

I chose the second option as it is reasonably simpler and the overhead for compiling a pattern of this size is rather small.